modules/affinity/src/main/java/org/apache/ignite/internal/affinity/AffinityService.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.affinity;
+
+import java.util.Collection;
+import java.util.List;
+import org.apache.ignite.network.ClusterNode;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Stateless affinity service that produces helper methods for an affinity assignments calculation.
+ */
+public class AffinityService {
+    /**
+     * Calculates affinity assignments.
+     *
+     * @param partitions Partitions count.
+     * @param replicas Replicas count.
+     * @return List nodes by partition.
+     */
+    public static List<List<ClusterNode>> calculateAssignments(
```

Comment:
What does service actually do?
Let's make method non-static or rename class to AffinityUtils?

Reply:
It calculates an assignment.
What's the point of making it non-static?
The same is about SchemaService (ex SchemaManager).

Reply:
I've just confused with naming. Its just a util method.

Reply:
Renamed to AffinityUtils, same for SchemaService->SchemaUtils.